### DISCUSSION TITLE
Kakao Map Development Progress

### DISCUSSION THREAD

**Discussion by mym0404** [mym0404/react-native-kakao#38] [Announcements]
> 25-03-19

Development of the map package is temporarily suspended due to military service.

======

Supporting Map on Android is not difficult. In fact, I fully support [react-native-naver-map](https://github.com/mym0404/react-native-naver-map) on Android and iOS in New and Old Arch, so I can use React Native's Map component to some extent. I have experience in porting.

But the problem is iOS. Currently Fabric is not fully compatible with Swift.

The Kakao Map V2 iOS SDK supports [Swift](https://devtalk.kakao.com/t/obj-c-kakao-sdk-2-0/116601) fully. I succeeded mount KakaoMap V2 in iOS with Objective-C++ but there are some lack feature in Objc.

Discussed in #28, I can separate codes in podspec with subsepc and supports C++ and Swift both in project. But I don't want to do that at now.

At [here](https://github.com/reactwg/react-native-new-architecture/discussions/143#discussioncomment-6680743). It seems Swift support in Fabric is not yet available.

The project can be delayed until after mature Fabric + Swift interop or we decide to just implement map with WebView.

Any idea is welcome.

**Comment by l2hyunwoo** [comment]
Hello, I think [nitro module](https://nitro.margelo.com/docs/what-is-nitro) is a good solution for this problem.

It seems much experimental, but there're no better solutions than this. I'd like to ask your opinion. Thanks!

**Reply by mym0404** [reply]
I've been keeping an eye on this project(nitro) and it looks nice.
I was alerted that author of nitro introduced it in conferences and it seems mature than past.

Previously, I selected Expo module as an alternative(Swift + Kotlin easy syntax) but Nitro seems deserve to compare.

But I don't have much time to support Map at now.

**Reply by mym0404** [reply]
Nitro is a proper solution for turbo module(only for simple utility functions) but [View Component](https://nitro.margelo.com/docs/view-components) yet. Furthermore, the main Bottleneck of implementing map component in react native was how to organize map API more developer friendly not speed.
